packages/eas-cli/src/commands/channel/list.ts

```diff
@@ -78,27 +31,9 @@ export default class ChannelList extends EasCommand {
     const exp = getExpoConfig(projectDir);
     const projectId = await getProjectIdAsync(exp);
 
-    const getAllUpdateChannelForAppResult = await getAllUpdateChannelForAppAsync({
-      appId: projectId,
+    await listAndRenderChannelsOnAppAsync({
+      projectId,
+      paginatedQueryOptions,
     });
-    const channels = getAllUpdateChannelForAppResult.app?.byId.updateChannels;
-    if (!channels) {
-      throw new Error(`Could not find channels on project with id ${projectId}`);
-    }
-
-    if (jsonFlag) {
-      printJsonOnlyOutput(channels);
```

Comment:
This change will make it very hard to make runAsync return value that should be printed in json only mode, as proposed here https://github.com/expo/eas-cli/pull/1360/files

You should probably discuss that with @wschurman to avoid large refactors back and forth.

Reply:
I wouldn't worry too much. I think my proposed change would require significant refactoring either way so I'd say it's fine to move forward with this.